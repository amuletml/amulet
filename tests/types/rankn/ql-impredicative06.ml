let auto (x : forall 'a. 'a -> 'a) : forall 'b. 'b -> 'b = x
let choose (x : 'a) (_ : 'a) = x

let id x = x

(* This /would/ work, as in the paper, but amulet deeply instantiates
 * the type of auto to be the same as auto' in ql-impredicative05.ml.
 *
 * Darn. *)
let t = choose id auto
