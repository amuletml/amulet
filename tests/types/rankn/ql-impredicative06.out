ql-impredicative06.ml[10:16 ..10:17]: error (E2001)
   │ 
10 │ let t = choose id auto
   │                ^^
  Couldn't match actual type 'b -> 'b
    with the type expected by the context, forall 'a. 'a -> 'a

  • Did you forget some of the arguments to a function?
